docs/HOWTO -- NVIDIA AGX ORIN  on-device kernel build.md

```diff
@@ -0,0 +1,112 @@
+Typicaly script is run as:
+```
+   $  ./nvbuild.sh -o $PWD/kernel_out
+```
+The result is stored in the ./kernel_out directory.
+
+6. Default kernel is ./kernel/kernel-5.10/arch/arm64/configs/tegra_defconfig
+7. I modified the ./nvbuild script adding an additional menuconfig phase so I can make slight modifications to the kernel config before  building:
+```
+
+        "${MAKE_BIN}" -C "${source_dir}" ARCH=arm64 \
+                LOCALVERSION="-tegra" \
+                CROSS_COMPILE="${CROSS_COMPILE_AARCH64}" \
+                "${O_OPT[@]}" "${config_file}"
+
+ +       "${MAKE_BIN}" -C "${source_dir}" ARCH=arm64 \
+ +               LOCALVERSION="-tegra" \
+ +               CROSS_COMPILE="${CROSS_COMPILE_AARCH64}" \
+ +               "${O_OPT[@]}"  \
+ +               --output-sync=target menuconfig
+ +
+        "${MAKE_BIN}" -C "${source_dir}" ARCH=arm64 \
+                LOCALVERSION="-tegra" \
+                CROSS_COMPILE="${CROSS_COMPILE_AARCH64}" \
+                "${O_OPT[@]}" -j"${NPROC}" \
+                --output-sync=target Image
+
+```
+
+8. To install build results I've developed another script, that installs kernel, modules and compiled DTBs in on the device:
+```
+./nvinstall.sh:
+
+#!/bin/bash
+
+# install kernel
+pushd kernel_out && sudo make modules_install &&  sudo make install && popd
+
+# install additional modules
+pushd ~/RTL88x2BU-Linux-Driver/ && make && sudo make install && popd
+
+# flash DTB
+pushd kernel_out && \
+sudo dd if=./arch/arm64/boot/dts/nvidia/tegra234-p3701-0000-p3737-0000.dtb of=/dev/mmcblk0p3 && \
+sudo dd if=./arch/arm64/boot/dts/nvidia/tegra234-p3701-0000-p3737-0000.dtb of=/dev/mmcblk0p6 && \
+popd
+``
+
+9. Root DTS for Nvidia Jetson ORIN is ./hardware/nvidia/platform/t23x/concord/kernel-dts/tegra234-p3701-0000-p3737-0000.dts. 
+
+Please mind it is out of the kernel source tree! But it is recompiled every time you build the kernel.
+
+10. Before installing the newly built kernel, update the bootloader config as in my example:
+```
+TIMEOUT 60
+DEFAULT primary
+
+MENU TITLE L4T boot options
+
+LABEL primary
+      MENU LABEL primary kernel
+      LINUX /boot/vmlinuz-5.10.104-tegra
+      INITRD /boot/initrd.img-5.10.104-tegra
+      APPEND ${cbootargs} root=/dev/mmcblk0p1 rw rootwait rootfstype=ext4 mminit_loglevel=4 console=ttyTCU0,115200 console=tty0 firmware_class.path=/etc/firmware fbcon=map:0 net.ifnames=0 
+
+ LABEL backup
+    MENU LABEL backup kernel
+    LINUX /boot/Image
+    INITRD /boot/initrd
+    APPEND ${cbootargs} root=/dev/mmcblk0p1 rw rootwait rootfstype=ext4 mminit_loglevel=4 console=ttyTCU0,115200 console=tty0 firmware_class.path=/etc/firmware fbcon=map:0 net.ifnames=0
+```
+
+So after ./nvinstall.sh you'll have your new kernel as primary and system default kernel as backup for emergency reasons.
+
+
+Thats all for today :-)
```

Comment:
I love it =)
But let's leave it for unofficial documentation.

Reply:
Nix actually generates these entries automatically with `nixos-rebuild switch` when the kernel is built the nix-way.